src/common/AndroidUtils.ts

```diff
@@ -589,18 +588,44 @@ export class AndroidSDKUtils {
         return `'${pathName}'`;
     }
 
-    private static enableHWKeyboardForEmulator(
+    private static updateEmulatorConfig(
         emulatorName: string
     ): Promise<boolean> {
-        return new Promise((resolve, reject) => {
-            try {
-                shell
-                    .echo('hw.keyboard=yes')
-                    .toEnd(`~/.android/avd/${emulatorName}.avd/config.ini`);
+        return new Promise(async (resolve, reject) => {
+            const config = await AndroidSDKUtils.readEmulatorConfig(
+                emulatorName
+            );
+            if (config.size === 0) {
+                // If we cannot edit the AVD config, fail silently.
+                // This will be a degraded experience but should still work.
                 resolve(true);
-            } catch (error) {
-                reject(error);
+                return;
             }
+
+            // Utilize hardware.
+            config.set('hw.keyboard', 'yes');
```

Comment:
@brandonpage Couldn't we add the property set to a config file instead? i.e. a config file with the overrides. In that way we do not have to make code changes for addl. keys and values.

Reply:
Yeah, I was planning on doing that but it turns out a few of the value need to computed at runtime.  I could add 5/7 of them to a config file.